app/services/get_next_sequence_counter.service.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+/**
+ * @module GetNextSequenceCounterService
+ */
+
+const { SequenceCounterModel } = require('../models')
+
+/**
+ * Returns the next bill_run_number for the given region and regime
+ *
+ * The bill run number in the sequence_counters table is the last number issued
+ * Therefore, we increment it by 1 and return the new number
+ *
+ * If an invalid region & regime pair is supplied, an Objection NotFoundError is thrown
+ *
+ * @param {string} regimeId Id of the regime to get the next counter for
+ * @param {string} region The region to get the next counter for
+ * @returns {integer} Value of the next counter
+ */
+class GetNextSequenceCounterService {
```

Comment:
Two things to look at here.

## Purpose of service

I'd argue that the service is specifically for handling everything to do with handling the next *bill run number* rather than a generic sequence counter. So I would argue we should have a more descriptive name for the service that encompasses this.

We know we'll need other *numbers* handled in future changes but I foresee those also being specific services with perhaps the bulk of the functionality moved to a `BaseSequenceThingyMeBobService`. But let's look at that when we get there.

## Avoid if possible get/set prefixes

Some languages like Ruby are built such that the need for `getFoo()` and `setFoo()` is completely unnecessary. This is because it makes a distinction between `foo` the property and `foo()` as a getter and setter method through other mechanisms.

A language like Java which doesn't make the distinction resorted to `getFoo()` and `setFoo()` for controlling access to the property `foo`, though even that is changing.

A popular argument now is that by avoiding prefixes like **get**, **set** and **is** you force yourself to come up with a more meaningful name that expresses better the intent of the logic underneath.

Where `get` and `set` is used it should follow certain rules. For example

- `getFoo()` should _only_ returning a private internal property directly and perform no other logic
- `setFoo()` should _only_ update a private internal property and perform no other logic

So when it comes to renaming this service we also need to look to drop the `Get` prefix.

Happy to discuss further and provide some resources if it will help.